airflow/hooks/S3_hook.py

```diff
@@ -53,7 +54,7 @@ def check_for_bucket(self, bucket_name):
         try:
             self.get_conn().head_bucket(Bucket=bucket_name)
             return True
-        except:
+        except Exception:
```

Comment:
This is bad, because it will also fail on `botocore.exceptions.NoCredentialsError: Unable to locate credentials`. Maybe make this a bit more elaborate: https://github.com/boto/boto3/blob/master/docs/source/guide/migrations3.rst#accessing-a-bucket

Reply:
Hey @Fokko , @kaxil , I revert these two lines for now as I don't have much context on what exceptions / errors got thrown by the functions. I think it would better to leave it to the S3 hook expert :)